tests/transaction_tests.cpp

```diff
@@ -100,22 +100,71 @@ TEST_CASE("Transaction guard") {
 
     storage.insert(Object{0, "Jack"});
 
-    SECTION("insert, call make a storage to call an exception and check that rollback was fired") {
-        auto countBefore = storage.count<Object>();
-        try {
-            auto guard = storage.transaction_guard();
+    struct ExceptionMatcher : Catch::Matchers::MatcherGenericBase {
+        std::string what;
+        std::error_code errorCode;
 
-            storage.insert(Object{0, "John"});
+        ExceptionMatcher(std::string what, std::error_code errorCode) :
+            what(std::move(what)), errorCode(std::move(errorCode)) {}
 
-            storage.get<Object>(-1);
+        bool match(const std::system_error& other) const {
+            const bool whatIsEqual = std::strcmp(other.what(), this->what.c_str()) == 0;
+            return whatIsEqual && other.code() == this->errorCode;
+        }
 
-            REQUIRE(false);
-        } catch(const std::system_error& e) {
-            REQUIRE(e.code() == orm_error_code::not_found);
-            auto countNow = storage.count<Object>();
+        std::string describe() const override {
+            std::stringstream ss;
+            const std::string message = get_orm_error_category().message(this->errorCode.value());
+            ss << "Exception with code: " << message << ", what: " << this->what;
+            return ss.str();
+        }
+    };
 
-            REQUIRE(countBefore == countNow);
+    const ExceptionMatcher notFoundExceptionMatcher("Not found", make_error_code(orm_error_code::not_found));
+    SECTION("insert, call make a storage to call an exception and check that rollback was fired") {
```

Comment:
BTW, this section within this test case replaces the same section in test case "transaction_rollback" some lines above, so the old one can be removed AFAIK?

Reply:
agree, I removed `TEST_CASE("transaction_rollback")`